store/src/main/java/org/apache/rocketmq/store/CommitLog.java

```diff
@@ -206,6 +208,12 @@ else if (!dispatchRequest.isSuccess()) {
             this.mappedFileQueue.setFlushedWhere(processOffset);
             this.mappedFileQueue.setCommittedWhere(processOffset);
             this.mappedFileQueue.truncateDirtyFiles(processOffset);
+
+            // Clear ConsumeQueue redundant data
+            if (isDamaged && maxPhyOffsetOfConsumeQueue >= processOffset) {
```

Comment:
It will be better if you can abstract this truncate logic and put it in the recover() method, because this problem exists no matter it is normal recovery or abnormal recovery.